contracts/NFTPool.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.7.5;
+pragma experimental ABIEncoderV2;
+
+import {DaiPool, Erc20Pool, ReceiverWeight, Dai} from "./Pool.sol";
+import {IERC721} from "openzeppelin-contracts/token/ERC721/IERC721.sol";
+
+/// @notice NFT pool contract to support streaming based on NFT ownership
+/// A NFT can be a sender or a receiver, a unique id is generated based on
+/// NFT registry address and the tokenId
+contract NFTPool is Erc20Pool {
```

Comment:
If it works on Dai, shouldn't it inherit `DaiPool`? We'd get support for quasi-ERC-712 signatures for free.

Reply:
Yes, I will add support for that as well together with the DAI `permit`. I already tried but sadly it resulted in too many local variables

Reply:
What do you mean? Have you hit a limitation of the arguments? Don't worry about gas price of making a call with such a long list of arguments, shaving off a single separate call to `approve` will immediately make it up for the user.